packages/store/addon/-private/system/fetch-manager.ts

```diff
@@ -0,0 +1,554 @@
+
+    if (recordData.isNew()) {
+      operation = 'createRecord';
+    } else if (recordData.isDeleted()) {
+      operation = 'deleteRecord';
+    } else {
+      operation = 'updateRecord';
+    }
+    let internalModel = snapshot._internalModel;
+    let modelName = snapshot.modelName;
+    let store = this._store;
+    let modelClass = store.modelFor(modelName);
+
+    assert(`You tried to update a record but you have no adapter (for ${modelName})`, adapter);
+    assert(
+      `You tried to update a record but your adapter (for ${modelName}) does not implement '${operation}'`,
+      typeof adapter[operation] === 'function'
+    );
+
+    let promise = Promise.resolve().then(() => adapter[operation](store, modelClass, snapshot));
+    let serializer = serializerForAdapter(store, adapter, modelName);
+    let label = `DS: Extract and notify about ${operation} completion of ${internalModel}`;
+
+    assert(
+      `Your adapter's '${operation}' method must return a value, but it returned 'undefined'`,
+      promise !== undefined
+    );
+
+    promise = guardDestroyedStore(promise, store, label);
+    promise = _guard(promise, _bind(_objectIsAlive, internalModel));
+
+    promise = promise.then(
+      adapterPayload => {
+        let payload, data, sideloaded;
+        if (adapterPayload) {
+          payload = normalizeResponseHelper(
+            serializer,
+            store,
+            modelClass,
+            adapterPayload,
+            snapshot.id,
+            operation
+          );
+          return payload;
+        }
+      },
+      function (error) {
+        if (error instanceof InvalidError) {
+          let parsedErrors = serializer.extractErrors(store, modelClass, error, snapshot.id);
+          throw { error, parsedErrors };
+        } else {
+          throw { error }
+        }
+      },
+      label
+    );
+    resolver.resolve(promise);
+  }
+
+
+  /**
+    This method is called at the end of the run loop, and
+    flushes any records passed into `scheduleSave`
+
+    @method flushPendingSave
+    @private
+  */
+  _flushPendingSaves() {
+    let pending = this._pendingSave.slice();
+    this._pendingSave = [];
+    for (var i = 0, j = pending.length; i < j; i++) {
+      let pendingItem = pending[i];
+      this._flushPendingSave(pendingItem);
+    }
+  }
+
+  scheduleFetch(identifier: RecordIdentifier, options: any, shouldTrace: boolean): Promise<any> {
+    // TODO Probably the store should pass in the query object
+
+
+    //debugger
+    let query: FindRecordQuery = {
+      'op': 'findRecord',
+      recordIdentifier: identifier,
+      options
+    }
+
+    let queryRequest: Request = {
+      data: [ query ]
+    }
+
+    // We already have a pending fetch for this
+    if (this._pendingFetch.has(identifier.type) && this._pendingFetch.get(identifier.type).find((fetch) => fetch.identifier.id === identifier.id)) {
```

Comment:
Double-check that `[].find()` is ok for IE11 use
<img width="443" alt="Screen Shot 2019-06-28 at 4 28 34 PM" src="https://user-images.githubusercontent.com/558005/60376156-c9822c00-99c1-11e9-8aef-c32d7ab9919c.png">